src/iddata/loader.py

```diff
@@ -404,6 +404,82 @@ def load_nhsn_from_nhsn(self, disease="flu", rates=True, as_of=None, drop_pandem
     dat["source"] = "nhsn"
     return dat
 
+  def load_nssp(self, disease="flu", drop_pandemic_seasons=True, as_of=None):
+    ## check disease is one of "flu", "covid" or "rsv"
+    valid_diseases = ["flu", "covid", "rsv"]
+    if disease not in valid_diseases:
+        raise ValueError("For NSSP data, the only supported diseases are 'flu', 'covid' and 'rsv'.")
+    
+    if as_of is None:
+      as_of = datetime.date.today()
+    
+    if isinstance(as_of, str):
+      as_of = datetime.date.fromisoformat(as_of)
+
+    ## for now, only support loading NSSP data from CDC for as_of dates on or after 2025-09-17
+    if as_of < datetime.date.fromisoformat("2025-09-17"):
+      raise NotImplementedError("Functionality for loading NSSP data with specified as_of date is not implemented.")
+    else:
```

Comment:
Python FYI: if your `if` causes a `raise`, then you don't typically use an `else` and instead just let the program flow. No change required, though.